### PR TITLE
FOGL-646 asset browser with storage engine

### PR DESCRIPTION
**Pull request by ashish-ScaleDB**
**assets readings browser**
   *  - [x]  /foglamp/asset
   *  - [x]  /foglamp/asset/{asset_code}
   *  - [x]  /foglamp/asset/{asset_code}/{reading}
   * - [x]  /foglamp/asset/{asset_code}/{reading}/summary
    * - [x]  /foglamp/asset/{asset_code}/{reading}/series - ~Blocked due to [FOGL-668](https://scaledb.atlassian.net/browse/FOGL-668)~

### CODE REVIEW

File: src/python/foglamp/core/api/browser.py [hunk 1]
```diff
@@ -70,77 +63,64 @@ def setup(app):
...
+    # PayloadBuilder().AGGREGATE(["count", "*"]).GROUP_BY('asset_code')
 
-    # Select the assets from the readings table
-    rows = await conn.fetch(
-        'SELECT asset_code, count(*) FROM readings GROUP BY asset_code')
-    columns = ('asset_code', 'count')
-    results = []
-    for row in rows:
-        results.append(dict(zip(columns, row)))
+    agg = {"operation": "count", "column": "*", "alias": "Count"}
```
Comment (praveen-garg) [inline src/python/foglamp/core/api/browser.py]: agg > aggregate

File: src/python/foglamp/core/api/browser.py [hunk 2]
```diff
@@ -70,77 +63,64 @@ def setup(app):
...
+    d = OrderedDict()
+    d['aggregate'] = agg
+    d['group'] = "asset_code"
```
Comment (praveen-garg) [inline src/python/foglamp/core/api/browser.py]: payload =  json.dumps(d)

File: src/python/foglamp/core/api/browser.py [hunk 3]
```diff
@@ -70,77 +63,64 @@ def setup(app):
...
 async def asset(request):
-    """
-    Browse a particular asset for which we have recorded readings and
+    """ Browse a particular asset for which we have recorded readings and
     return a readings with timestamps for the asset. The number of readings
     return is defaulted to a small number (20), this may be changed by supplying
     the query parameter ?limit=xx&skip=xx
 
-    Return the result of the Postgres query 
+    Return the result of the Postgres query
```
Comment (praveen-garg) [inline src/python/foglamp/core/api/browser.py]: remove postgres

File: src/python/foglamp/core/api/browser.py [hunk 4]
```diff
@@ -70,77 +63,64 @@ def setup(app):
...
-
-    query += _where_clause(request)
+    # TODO: FOGL-637, 640
+    timestamp = {"column": "user_ts", "format": __TIMESTAMP_FMT, "alias": "timestamp"}
+    d = OrderedDict()
+    d['return'] = [timestamp, "reading"]
+    d['where'] = {"column": "asset_code", "condition": "=", "value": asset_code}
+    _and = _where_clause(request)
+    if len(_and) > 0:
+        d['where'].update(_and)
```
Comment (praveen-garg) [inline src/python/foglamp/core/api/browser.py]: _and_where = _where_clause(request)

File: src/python/foglamp/core/api/browser.py [hunk 5]
```diff
@@ -70,77 +63,64 @@ def setup(app):
...
-        jrow = {'timestamp': row['timestamp'], 'reading': json.loads(row['reading'])}
-        results.append(jrow)
+    _storage = connect.get_storage()
+    results = _storage.query_tbl_with_payload('readings', json.dumps(d))
 
-    # Close the connection.
-    await conn.close()
-
-    return web.json_response(results)
+    return web.json_response(results['rows'])
```
Comment (praveen-garg) [inline src/python/foglamp/core/api/browser.py]: in case of error : return web.json_response(results['message']) Here HTTP status code is 200 itself

Comment (ashish-ScaleDB) [reply src/python/foglamp/core/api/browser.py]: We will handle this as separately task. for all end-points

Comment (MarkRiddoch) [reply src/python/foglamp/core/api/browser.py]: Please add task/TODO for this. Then we can approve and get this merged in.

**Pull request merged**